### PR TITLE
Adds the new metadata changes

### LINKED ISSUE #60: Add sanity metadata checks to SDK
On two levels:
1 - metadata may not be syntactically valid;
2 - there is currently no input or output sanity check against metadata which makes no sense in the context (eg scalar market may have categories)

See #59

### LINKED ISSUE #89: Use new market metadata type
The following pull request changes the market metadata type: https://github.com/zeitgeistpm/zeitgeist/pull/120

Please prepare the SDK to be able to submit only valid ipfs multihashes to the `create_*_market` functions **(1)** and to distinguish between old `CIDv0` ipfs metadata and new `MultiHash` metadata when querying the chain storage **(2)**. Precisely:

**(1)** When a new file should be added to ipfs, ensure that it uses CIDv1 and SHA3-384 hash algorithm (this implies CIDv1). Using the ipfs client that can be achieved by using following command (The API offers similar functions):
```sh
ipfs add --cid-version 1 --hash sha3-384 test.json
```

Use the next command to extract a hexadecimal string from the encoded CIDv1 string that represents the complete multihash (use `-b="identity"` to retrieve 8-bit binary [encoder and decoder keeps data unmodified]):
```sh
ipfs cid format -b="f" -f "%M " -- bafkrkmcpdvpn5lz3l2chedmrohfjp2l4wa6ozmq5mf4kkr2j3sjqed2h7dsr2ifcvk3q6x2pwgoiwgldvcra
```
> 15304f1d5edeaf3b5e84720d9171ca97e97cb03cecb21d6178a54749dc93020f47f8e51d20a2aab70f5f4fb19c8b1963a8a2

The result must be put into `MultiHash::Sha3-384` enum before submitting it to the `create_+_market` functions.

**(2)** When market metadata is queried from the chain storage, a vector (arbitrary length byte array) is returned. You can check  whether it is a raw `MultiHash::Sha3-384` or a valid completely encoded CIDv0 (or newer) string by checking the first byte:

```rust
if metadata[0] = 0x15 {
    // Raw multi hash, needs to be encoded properly:
   valid_cid = concat(f0155, hex(metadata[0..50]))
} else {
   valid_cid = metadata
}
```

### PR DESCRIPTION
**Pull request by lsaether**
closes #89 
closes #60

### CODE REVIEW

**Review by sea212 — CHANGES_REQUESTED** [review]
Thanks for implementing this so fast. There is only one major issue regarding the encoding, otherwise it looks great.

File: packages/sdk/src/storage/ipfs.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    // Old way - backwards compatibility is fun
+    if (what.slice(2, 6) !== "1530") {
+      const str = hexToString(what);
+      return toString(concat(await all(this.client.cat(str))));
+    }
+
+    // New way
+    const cid = new CID("f0155" + what.slice(2));
+    const encoded = cid.toString("base32");
+    const data = await all(this.client.cat(encoded));
```
[the remark this answers is not in the document]
Comment (lsaether) [reply packages/sdk/src/storage/ipfs.ts:29]: Weird.. this way works for me in practice. Will double check it.

[the remark this answers is not in the document]
Comment (lsaether) [reply packages/sdk/src/storage/ipfs.ts:29]: CID starts as hex encoded, but is being turned to base 32 encoding in order to fetch it from IPFS. What you're saying is that we don't need to base32 encode, but can fetch it using the hex encoded CID?

[the remark this answers is not in the document]
Comment (sea212) [reply packages/sdk/src/storage/ipfs.ts:29]: Yes, `"f0155" + what.slice(2)` is a valid CID and does not demand further processing, if `what.slice(2)` is a valid hex representation of the multihash.
If `what.slice(2)` would be base32 encoded for example, the CID would be generated like this (pseudocode) `"b" + [0x01, 0x55].toString("base32") + what.slice(2)`.
.Sorry, my comment before this one was wrong in that "b" is only a valid prefix if the following data is already base32 encoded. Very interesting though that IPFS seems to be able to correctly interpret a correct CID that was additionally base32 encoded.

**Pull request merged**